### DISCUSSION TITLE
[Feature Request]Hide recommended videos from feed and related(next and similar) videos from player

### DISCUSSION THREAD

**Discussion by tokyocrib** [ReVanced/revanced-discussions#348] [Question]
Hide all recommended and related videos in feed from settings with options like "contents of main page" and "show 'next' and 'similar' videos" like in newpipe.

![Screenshot_2022-07-21-22-46-49-096_org.polymorphicshade.newpipe_1_1.jpg](https://user-images.githubusercontent.com/107816362/180274858-729bd4e9-e320-4b7b-9621-99d0000d4f93.jpg)

**Comment by oSumAtrIX** [comment]
Isn't this already a thing? @TheJeterLP

**Comment by TheJeterLP** [comment]
> Isn't this already a thing? @TheJeterLP 

No, recommendations on feed are still there

**Reply by tokyocrib** [reply]
Can option to toggle recommended and related videos be added in revanced?

**Comment by mayel** [comment]
+1 to both of these, especially the related/suggested videos that appear after/under a video, which can be a problem especially when kids watch YouTube, since very random videos can pop up there

**Comment by Shadow-302** [comment]
Helpful suggestion.

**Comment by DistroAgnostic** [comment]
Sad to know this feature still not added!

**Comment by Khiu** [comment]
This can now be done by going to settings > revanced > layout > custom filter and adding related_video there.

**Reply by Shibbyz** [reply]
Unfortunately, this does not work in 2024-07. I have tried both "related_video" and "related_videos" with no success. Perhaps custom filters just broke a while back and no one noticed?

**Reply by LisoUseInAIKyrios** [reply]
Custom filter still works. 

`related_video` still works and regular videos in the video player recommendations are hidden, but Shorts recommendations are not hidden so under the player is nothing but shorts.

**Reply by DistroAgnostic** [reply]
> Custom filter still works. 
> 
> `related_video` still works and regular videos in the video player recommendations are hidden, but Shorts recommendations are not hidden so under the player is nothing but shorts.

No you're wrong, what you're saying happens before & before that community posts shows up generally but now i do get video suggestions.

**Reply by DistroAgnostic** [reply]
> Unfortunately, this does not work in 2024-07. I have tried both "related_video" and "related_videos" with no success. Perhaps custom filters just broke a while back and no one noticed?

I noticed bro.

**Reply by oSumAtrIX** [reply]
Apart from the filter working as expected just like Liso said, blocking feed items like this is straining the API by the client bombarding the API with requests in attempt to load new feed items, until either the API or client is exhausted. What you are trying to do is everything but a good idea regardless, and it's good it doesn't work for you

**Reply by tikkamasalla** [reply]
> Apart from the filter working as expected just like Liso said, blocking feed items like this is straining the API by the client bombarding the API with requests in attempt to load new feed items, until either the API or client is exhausted. What you are trying to do is everything but a good idea regardless, and it's good it doesn't work for you

So do you work for YouTube? Why would you care if the api is being strained?

**Reply by LisoUseInAIKyrios** [reply]
It's more than that.  Your device will keep making the same requests in a loop and basically never stop.  Want battery drain and overheating?   It's not a viable option.

**Reply by DistroAgnostic** [reply]
> It's more than that.  Your device will keep making the same requests in a loop and basically never stop.  Want battery drain and overheating?   It's not a viable option.

Well, It's actually viable as I open youtube to watch a video, watch it then close but after getting suggestions i found myself stuck in loop for hours.

**Reply by DistroAgnostic** [reply]
> Apart from the filter working as expected just like Liso said, blocking feed items like this is straining the API by the client bombarding the API with requests in attempt to load new feed items, until either the API or client is exhausted. What you are trying to do is everything but a good idea regardless, and it's good it doesn't work for you

Is there any other way than exhausting api, I mean can't we simply hide suggested videos like newpipe did? Just show a empty screen there or add some overlay there to cover it or something.

**Reply by oSumAtrIX** [reply]
You can simply change the start page.

**Reply by Shibbyz** [reply]
Revanced Extended has this feature and it's working. I don't recall any battery drainage or overheating when I used it. What if they found a solution to the API problem everyone's talking about? Someone that understands this stuff should ask them.

**Reply by Shibbyz** [reply]
> You can simply change the start page.

No, that's not it. We're talking about the recommended videos that appear underneath the video player when you are watching a video.

**Reply by oSumAtrIX** [reply]
Enter fullscreen. What the fork does is straining the API like I said which won't be reflected here.

**Reply by LisoUseInAIKyrios** [reply]
Or swipe and pull down on the video title.  It will stay in portrait mode and everything except the video is hidden.

**Reply by Shibbyz** [reply]
> Or swipe and pull down on the video title. It will stay in portrait mode and everything except the video is hidden.

That would be like telling the heroin addict to just put the heroin in the cupboard and they can't use it anymore. Haha.
